gammapy/maps/map_dict.py

```diff
@@ -0,0 +1,139 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from collections import UserDict
+from collections.abc import MutableMapping
+from astropy.io import fits
+from gammapy.maps import Map
+from gammapy.utils.scripts import make_path
+
+__all__ = ["Maps"]
+
+
+class Maps(MutableMapping):
+    """A Dictionary containing Map objects sharing the same geometry.
+
+    This class simplifies handling and I/O of maps collections.
+
+    For maps with different geometries, use a regular dict.
+    """
+
+    def __init__(self, **kwargs):
+        self._geom = None
+        self._data = {}
+        for key, value in kwargs.items():
+            self.__setitem__(key, value)
+
```

Comment:
I think a method like `Maps.from_geom(geom, names=[])` could be useful to create an empty container to be filled with data later...I think there are many use case within Gammapy for this.

Reply:
Done. The function is `Maps.from_geom(geom, names, kwargs_list=None)` to have the possibility to pass the arguments.

Then it can be called:
```
    geom = WcsGeom.create(npix=5)
    names = ["map1", "map2", "map3"]
    kwargs_list = [
        {"unit":"cm2s", "dtype":"float64"},
        {"dtype" : "bool"},
        {"data" : np.arange(25).reshape(5,5)}
    ]

    maps = Maps.from_geom(geom, names, kwargs_list=kwargs_list)